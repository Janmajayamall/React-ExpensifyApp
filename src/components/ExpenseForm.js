import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';


export default class ExpenseForm extends React.Component{
    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note:props.expense ? props.expense.note : '',
            amount:props.expense ? (props.expense.amount/100).toString() : '',
            createdAt:props.expense ? moment(props.expense.createdAt) : moment(),
            calenderFocused: false,
            error: ''
        };

    }
    
  

    onDescriptionChange = (e) =>{
        const description = e.target.value;
        this.setState(()=>{
            return {
                description: description
            }
        })
    }

    onNoteChange = (e) =>{
        const note = e.target.value;
        this.setState(()=>{
            return{
                note: note
            }
        })
    }

    onAmountChange = (e) =>{
        const amount = e.target.value;

    if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>{
                return{
                    amount: amount
                }
            })
        }
       
    }

    onDateChange = (createdAt) =>{
        if(createdAt){
            this.setState(()=>({createdAt:createdAt}));

        }
    }

    onFocusChange = ({focused}) =>{
        this.setState(()=>({calenderFocused:focused}));
    }

    onSubmit = (e) =>{
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=>{
                return{
                    error: 'Description and Amount of the expense are required fields'
                }
            });
        }else{
            this.setState(()=>{
                return{
                    error:''
                }
            });
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount)*100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            });
        }
    }

    render(){
        return(
             <form className='form' onSubmit = {this.onSubmit}>
                    {this.state.error && <p className='form__error'>{this.state.error}</p>}
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value = {this.state.description}
                        className = "text-input"
                        onChange = {this.onDescriptionChange}

                    />
                    <input
                        type="text"
                        placeholder="Amount"
                        value = {this.state.amount}
                        className = "text-input"
                        onChange = {this.onAmountChange}

                    />
                    <textarea
                        className='text-area'
                        placeholder="If you want to add any notes on the expense, add them here!"
                        value = {this.state.note}
                        onChange={this.onNoteChange}>
                    </textarea>
                    <SingleDatePicker
                        date= {this.state.createdAt}
                        onDateChange = {this.onDateChange}
                        focused = {this.state.calenderFocused}
                        onFocusChange = {this.onFocusChange}
                        numberOfMonths= {1}
                        isOutsideRange= {()=>(false)}
                    />
                    <div>
                    <button className='button'>
                        Add Expense
                    </button>
                    </div>
                </form>
        );
    }
}