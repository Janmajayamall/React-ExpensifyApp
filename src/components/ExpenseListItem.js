import React from 'react';
import { connect } from 'react-redux';
import {removeExpense} from './../actions/expenses.js';
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'


const ExpenseListItem = ( {dispatch ,id ,description, amount, createdAt}) =>{
    return(
        <div>
            <Link to={`/edit/${id}`} >{description}</Link>
            <p>Amount: {amount/100} and createdAt: {createdAt}</p>

        </div>
    );
}   

export default ExpenseListItem;