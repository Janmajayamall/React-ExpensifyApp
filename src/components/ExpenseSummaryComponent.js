import React from 'react';
import getTotalExpenses from './../selectors/expensesTotal.js';
import numeral from 'numeral'


export default class ExpenseSummaryComponent extends React.Component{
    render(){
        const expenseSING = this.props.expenses.length === 1 ? 'expense' : 'expenses';
        return(
            <div>
                <p>Viewing {this.props.expenses.length} {expenseSING} totalling {numeral(getTotalExpenses(this.props.expenses)/100).format('$0,0.00')}</p> 
            </div>
        );
    }
}

