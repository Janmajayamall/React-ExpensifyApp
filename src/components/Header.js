import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from './../actions/auth';

   

export const Header = (props)=>(
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName = "is-active" exact={true}> Go Home </NavLink>
        <NavLink to="/create" activeClassName = "is-active"> Add Expense</NavLink>
        <NavLink to="/help" activeClassName = "is-active"> Help </NavLink>
        <button onClick={props.startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) =>{
    return {
        startLogout: () =>{dispatch(startLogout())}
    }
}

export default connect(undefined, mapDispatchToProps)(Header);