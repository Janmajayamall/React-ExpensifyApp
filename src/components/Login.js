import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from './../actions/auth.js'

export const LoginPage = (props) =>{
    return(
        <div className = 'box-layout'>
            <div className = 'box-layout__box'>
                <h1 className='box-layout__title'>SpendAdd</h1>
                <p>Don't worry for expenses from now onwards</p>
                <button className = 'button' onClick={props.startLogin}>Google Login</button>
            </div>
        </div>
    );
}

const mapDispatchToProps = (dispatch) =>{
    return {
        startLogin: () =>{dispatch(startLogin())}
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage);
