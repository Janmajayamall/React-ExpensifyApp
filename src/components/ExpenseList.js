import React from 'react';
import ExpenseListItem from './ExpenseListItem';



export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? <p>No expense to show</p>
                : props.expenses.map((expense)=>{
            return(
                <ExpenseListItem key = {expense.id} {...expense}/>
                    );       
            })
        }
    </div>
);



export default ExpenseList;