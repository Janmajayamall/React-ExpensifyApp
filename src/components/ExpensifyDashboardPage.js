import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';
import {ExpenseList} from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters.js';
import ExpenseSummaryComponent from './ExpenseSummaryComponent.js';
import { connect } from 'react-redux';
import selectExpenses from './../selectors/expenses.js';


export const ExpensifyDashboardPage = (props) =>{

        return (
            <div>
                <ExpenseSummaryComponent expenses = {props.expenses} />
                <ExpenseListFilters/>
                <ExpenseList expenses = {props.expenses}/>
            </div>
        );}


const mapStateToProps = (state, props)=>{
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}
export default connect(mapStateToProps)(ExpensifyDashboardPage);