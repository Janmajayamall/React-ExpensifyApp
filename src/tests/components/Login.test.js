import {LoginPage} from './../../components/Login';
import {shallow} from 'enzyme';
import React from 'react';



test('Login Page should render properly', ()=>{
    const wrapper = shallow(<LoginPage startLogin={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();
})

test('Should call startLogin function properly', ()=>{
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});