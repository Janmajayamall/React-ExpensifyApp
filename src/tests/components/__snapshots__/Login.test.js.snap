// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render properly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      A new way to have your expenses under control
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login in with Google
    </button>
  </div>
</div>
`;
